receiver/windowsperfcountersreceiver/config.go

```diff
@@ -25,14 +25,40 @@ import (
 type Config struct {
 	scraperhelper.ScraperControllerSettings `mapstructure:",squash"`
 
-	PerfCounters []PerfCounterConfig `mapstructure:"perfcounters"`
+	MetricMetaData []MetricConfig      `mapstructure:"metric_metadata"`
+	PerfCounters   []PerfCounterConfig `mapstructure:"perfcounters"`
 }
 
 // PerfCounterConfig defines configuration for a perf counter object.
 type PerfCounterConfig struct {
-	Object    string   `mapstructure:"object"`
-	Instances []string `mapstructure:"instances"`
-	Counters  []string `mapstructure:"counters"`
+	Object    string          `mapstructure:"object"`
+	Instances []string        `mapstructure:"instances"`
+	Counters  []CounterConfig `mapstructure:"counters"`
+}
+
+// MetricsConfig defines the configuration for a metric to be created.
+type MetricConfig struct {
+	MetricName  string      `mapstructure:"metric_name"`
+	Unit        string      `mapstructure:"unit"`
+	Description string      `mapstructure:"description"`
+	Gauge       GaugeMetric `mapstructure:"gauge"`
+	Sum         SumMetric   `mapstructure:"sum"`
+}
+
+type GaugeMetric struct {
+	ValueType string `mapstructure:"value_type"`
+}
+
+type SumMetric struct {
+	ValueType   string `mapstructure:"value_type"`
```

Comment:
I'm not a huge fan of diverging from the structure in `mdatagen`. I think it's likely that we eventually have some direct interoperability between that package and this config. In that case, this would require a breaking change.

That said, a couple specifics:
1. I think we can remove the value type altogether. If we later switch to using the mdatagen struct, the value will reappear, but won't do anything. This is not a breaking change so seems fine.
2. Even though gauge doesn't have any details, it could eventually. In any case, if we make it the default as suggested elsewhere, then users won't even have to specify it.

Reply:
Updated